### ISSUE TITLE
Proxy Switch is not working properly

### ISSUE THREAD

**Issue by shirshak55** [httptoolkit/httptoolkit#168]
When I change proxies its not being changed and sometimes it uses an old proxy sometimes doesn't use a proxy. It's creating a lot of confusion. It seems a bit buggy to me.

I will see where is this problem coming from.

**Comment by pimterry** [comment]
You mean when changing the upstream proxy settings in the HTTP Toolkit UI? That's interesting OK, good catch! If you could add some quick steps to reproduce the issue here once you've chased that down, then that'd be very useful, thanks.

**Comment by shirshak55** [comment]
@pimterry yep upstream proxy. Just try opening firefox and change the proxy and go to jsonip.com etc. I think it should reproduce easily. 

And try changing to another proxy again it wont change. I don't know why. However if I restart httptoolkit it works again.
